diff -urNp mcelog-1.0pre3_20120814_2.orig/mcelog.init mcelog-1.0pre3_20120814_2/mcelog.init
--- mcelog-1.0pre3_20120814_2.orig/mcelog.init	2014-03-21 11:10:05.391742495 -0400
+++ mcelog-1.0pre3_20120814_2/mcelog.init	2014-03-21 11:26:13.022369505 -0400
@@ -7,6 +7,9 @@
 # The paths are hardcoded and are not automatically adjusted
 # for different prefix
 #
+# chkconfig: 35 50 50
+# description: mcelog hardware error logging
+#
 ### BEGIN INIT INFO
 # Provides:		mcelog 
 # Default-Start:	3 5
