diff -urNp mcelog-1.0pre3_20120814_2.orig/intel.c mcelog-1.0pre3_20120814_2/intel.c
--- mcelog-1.0pre3_20120814_2.orig/intel.c	2014-03-21 14:39:25.027612913 -0400
+++ mcelog-1.0pre3_20120814_2/intel.c	2014-03-21 14:39:30.104599324 -0400
@@ -59,7 +59,7 @@ enum cputype select_intel_cputype(int fa
 			return CPU_NEHALEM;
 		else if (model == 0x2e || model == 0x2f)
 			return CPU_XEON75XX;
-		else if (model == 0x2a || model == 0x3a)
+		else if (model == 0x2a)
 			return CPU_SANDY_BRIDGE;
 		else if (model == 0x2d)
 			return CPU_SANDY_BRIDGE_EP;
diff -urNp mcelog-1.0pre3_20120814_2.orig/ivy-bridge.c mcelog-1.0pre3_20120814_2/ivy-bridge.c
--- mcelog-1.0pre3_20120814_2.orig/ivy-bridge.c	2014-03-21 14:39:25.027612913 -0400
+++ mcelog-1.0pre3_20120814_2/ivy-bridge.c	2014-03-21 14:44:08.673948184 -0400
@@ -76,6 +76,7 @@ static char *memctrl_1[] = {
 	[0x010] = "Uncorrected patrol scrub error",
 	[0x020] = "Corrected spare error",
 	[0x040] = "Uncorrected spare error",
+	[0x080] = "Corrected memory read error",
 	[0x100] = "iMC, WDB, parity errors",
 };
 
