diff -urNp mcelog-1.0pre3_20120814_2.orig/mcelog.c mcelog-1.0pre3_20120814_2/mcelog.c
--- mcelog-1.0pre3_20120814_2.orig/mcelog.c	2013-03-12 08:58:24.956875893 -0400
+++ mcelog-1.0pre3_20120814_2/mcelog.c	2013-03-12 09:09:33.870972014 -0400
@@ -501,9 +501,10 @@ int is_cpu_supported(void)
 			if (!strcmp(vendor,"AuthenticAMD")) {
 				if (family == 15)
 					cputype = CPU_K8;
-				if (family > 15)
+				if (family > 15) {
 					fprintf(stderr, "AMD Processor family %d: Please load edac_mce_amd module.\n", family);
 				return 0;
+				}
 			} else if (!strcmp(vendor,"GenuineIntel"))
 				cputype = select_intel_cputype(family, model);
 			/* Add checks for other CPUs here */	
