diff -urNp mcelog-1.0pre3_20120814_2.orig/intel.c mcelog-1.0pre3_20120814_2/intel.c
--- mcelog-1.0pre3_20120814_2.orig/intel.c	2013-08-14 11:53:13.815911637 -0400
+++ mcelog-1.0pre3_20120814_2/intel.c	2013-08-14 13:21:26.619124799 -0400
@@ -32,7 +32,8 @@ void intel_cpu_init(enum cputype cpu)
 {
 	if (cpu == CPU_NEHALEM || cpu == CPU_XEON75XX || cpu == CPU_INTEL ||
 	    cpu == CPU_SANDY_BRIDGE || cpu == CPU_SANDY_BRIDGE_EP ||
-	    cpu == CPU_IVY_BRIDGE || cpu == CPU_IVY_BRIDGE_EPEX)
+	    cpu == CPU_IVY_BRIDGE || cpu == CPU_IVY_BRIDGE_EPEX ||
+	    cpu == CPU_HASWELL)
 		memory_error_support = 1;
 }
 
@@ -66,6 +67,8 @@ enum cputype select_intel_cputype(int fa
 			return CPU_IVY_BRIDGE;
 		else if (model == 0x3e)
 			return CPU_IVY_BRIDGE_EPEX;
+		else if (model == 0x3c || model == 0x45 || model == 0x46)
+			return CPU_HASWELL;
 		if (model > 0x1a) {
 #if 0
 			Eprintf("Family 6 Model %x CPU: only decoding architectural errors\n",
diff -urNp mcelog-1.0pre3_20120814_2.orig/intel.c.orig mcelog-1.0pre3_20120814_2/intel.c.orig
--- mcelog-1.0pre3_20120814_2.orig/intel.c.orig	2013-08-14 11:53:13.815911637 -0400
+++ mcelog-1.0pre3_20120814_2/intel.c.orig	2013-08-14 11:53:06.358773056 -0400
@@ -22,6 +22,8 @@
 #include "nehalem.h"
 #include "memdb.h"
 #include "page.h"
+#include "sandy-bridge.h"
+#include "ivy-bridge.h"
 #include "xeon75xx.h"
 
 int memory_error_support;
@@ -29,7 +31,8 @@ int memory_error_support;
 void intel_cpu_init(enum cputype cpu)
 {
 	if (cpu == CPU_NEHALEM || cpu == CPU_XEON75XX || cpu == CPU_INTEL ||
-	    cpu == CPU_SANDY_BRIDGE || cpu == CPU_SANDY_BRIDGE_EP)
+	    cpu == CPU_SANDY_BRIDGE || cpu == CPU_SANDY_BRIDGE_EP ||
+	    cpu == CPU_IVY_BRIDGE || cpu == CPU_IVY_BRIDGE_EPEX)
 		memory_error_support = 1;
 }
 
@@ -59,15 +62,23 @@ enum cputype select_intel_cputype(int fa
 			return CPU_SANDY_BRIDGE;
 		else if (model == 0x2d)
 			return CPU_SANDY_BRIDGE_EP;
+		else if (model == 0x3a)
+			return CPU_IVY_BRIDGE;
+		else if (model == 0x3e)
+			return CPU_IVY_BRIDGE_EPEX;
 		if (model > 0x1a) {
+#if 0
 			Eprintf("Family 6 Model %x CPU: only decoding architectural errors\n",
 				model);
+#endif
 			return CPU_INTEL; 
 		}
 	}
 	if (family > 6) { 
+#if 0
 		Eprintf("Family %u Model %x CPU: only decoding architectural errors\n",
 				family, model);
+#endif
 		return CPU_INTEL;
 	}
 	Eprintf("Unknown Intel CPU type family %x model %x\n", family, model);
@@ -98,6 +109,12 @@ static int intel_memory_error(struct mce
 		case CPU_XEON75XX:
 			xeon75xx_memory_error(m, recordlen, channel, dimm);
 			break;
+		case CPU_SANDY_BRIDGE_EP:
+			sandy_bridge_ep_memerr_misc(m, channel, dimm);
+			break;
+		case CPU_IVY_BRIDGE_EPEX:
+			ivy_bridge_ep_memerr_misc(m, channel, dimm);
+			break;
 		default:
 			break;
 		} 
diff -urNp mcelog-1.0pre3_20120814_2.orig/intel.h mcelog-1.0pre3_20120814_2/intel.h
--- mcelog-1.0pre3_20120814_2.orig/intel.h	2013-08-14 11:53:13.815911637 -0400
+++ mcelog-1.0pre3_20120814_2/intel.h	2013-08-14 11:53:44.890489335 -0400
@@ -17,5 +17,6 @@ extern int memory_error_support;
 	case CPU_SANDY_BRIDGE_EP: \
 	case CPU_SANDY_BRIDGE: \
 	case CPU_IVY_BRIDGE: \
-	case CPU_IVY_BRIDGE_EPEX
+	case CPU_IVY_BRIDGE_EPEX: \
+	case CPU_HASWELL
 
diff -urNp mcelog-1.0pre3_20120814_2.orig/mcelog.c mcelog-1.0pre3_20120814_2/mcelog.c
--- mcelog-1.0pre3_20120814_2.orig/mcelog.c	2013-08-14 11:53:13.815911637 -0400
+++ mcelog-1.0pre3_20120814_2/mcelog.c	2013-08-14 11:53:44.891489353 -0400
@@ -227,6 +227,7 @@ static char *cputype_name[] = {
 	[CPU_SANDY_BRIDGE_EP] = "Sandy Bridge EP", /* Fill in better name */
 	[CPU_IVY_BRIDGE] = "Ivy Bridge", /* Fill in better name */
 	[CPU_IVY_BRIDGE_EPEX] = "Ivy Bridge EP/EX", /* Fill in better name */
+	[CPU_HASWELL] = "Haswell", /* Fill in better name */
 };
 
 static struct config_choice cpu_choices[] = {
@@ -262,6 +263,7 @@ static struct config_choice cpu_choices[
 	{ "ivybridge", CPU_IVY_BRIDGE }, /* Fill in better name */
 	{ "ivybridge-ep", CPU_IVY_BRIDGE_EPEX }, /* Fill in better name */
 	{ "ivybridge-ex", CPU_IVY_BRIDGE_EPEX }, /* Fill in better name */
+	{ "haswell", CPU_HASWELL }, /* Fill in better name */
 	{}
 };
 
diff -urNp mcelog-1.0pre3_20120814_2.orig/mcelog.h mcelog-1.0pre3_20120814_2/mcelog.h
--- mcelog-1.0pre3_20120814_2.orig/mcelog.h	2013-08-14 11:53:13.816911668 -0400
+++ mcelog-1.0pre3_20120814_2/mcelog.h	2013-08-14 11:53:44.892489371 -0400
@@ -117,6 +117,7 @@ enum cputype {
 	CPU_SANDY_BRIDGE_EP, 
 	CPU_IVY_BRIDGE, 
 	CPU_IVY_BRIDGE_EPEX, 
+	CPU_HASWELL,
 };
 
 enum option_ranges {
