diff -urNp mcelog-1.0pre3_20120814_2.orig/mcelog.c mcelog-1.0pre3_20120814_2/mcelog.c
--- mcelog-1.0pre3_20120814_2.orig/mcelog.c	2012-10-31 10:58:25.424368412 -0400
+++ mcelog-1.0pre3_20120814_2/mcelog.c	2012-10-31 10:58:53.279510730 -0400
@@ -501,7 +501,7 @@ int is_cpu_supported(void)
 			if (!strcmp(vendor,"AuthenticAMD")) {
 				if (family == 15)
 					cputype = CPU_K8;
-				if (family >= 15)
+				if (family > 15)
 					fprintf(stderr, "AMD Processor family %d: Please load edac_mce_amd module.\n", family);
 				return 0;
 			} else if (!strcmp(vendor,"GenuineIntel"))
