diff -urNp mcelog-1.0pre3_20120814_2.orig/mcelog.conf mcelog-1.0pre3_20120814_2/mcelog.conf
--- mcelog-1.0pre3_20120814_2.orig/mcelog.conf	2013-08-14 14:09:01.901049433 -0400
+++ mcelog-1.0pre3_20120814_2/mcelog.conf	2013-08-14 14:07:29.313333800 -0400
@@ -15,6 +15,10 @@
 #no-syslog = yes   (or no to disable)
 # when the option has a argument
 #logfile = /tmp/logfile
+
+# by default, disable extended error logging on newer Intel processors
+no-imc-log = yes
+
 # below are the options which are not command line options
 
 # Set CPU type for which mcelog decodes events:
